### PR TITLE
Fix sloshing electricity & enable SpeechVerb masking

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Pulsing the mic wire on an intercom will now use an appropriate verb for the chatlog message. Electricity can crackle, buzz, and screech.

## Why / Balance

electric_slosh
Previously, the message used the speech verb set of the person pulsing the wire. This looked silly in many cases.

## Technical details

VoiceMaskComponent now has a nullable SpeechVerb property. If not null, it will override the speech verbs used by the entity.

I hooked this up for the special case of radio messages* and the general use of speaking too, for completeness. Note that I didn't let the user modify this on wearable voicemasks since that could affect game balance (is catching people using the wrong speech verbs for the character they're disguised as intended gameplay?).

This also has potential for admin shenanigans - give a player a voicemask component and set the SpeechVerb to Canine and they'll bark woof and howl all their chat messages.

*Why is this a special case anyway? It probably should use TransformSpeakerNameEvent like normal speech.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by deltanedas** [comment]
> hooked this up for the special case of radio messages* and the general use of speaking too, for completeness. Note that I didn't let the user modify this on wearable voicemasks since that could affect game balance (is catching people using the wrong speech verbs for the character they're disguised as intended gameplay?).

hell no. if you can fool the entire station into hearing someone elses voice, how would it be instantly recognizable as a different species

i will make pr for selecting any player accent (including silicons for trolling on binary) after this :trollface:

File: Content.Server/Speech/Components/ListenWireAction.cs [hunk 1]
```diff
@@ -34,6 +41,7 @@ public override void Initialize()
 
         _wires = EntityManager.System<WiresSystem>();
         _chat = EntityManager.System<ChatSystem>();
+        _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
```
Comment (metalgearsloth) [inline Content.Server/Speech/Components/ListenWireAction.cs:44]: Don't these get dependency injected.

Comment (Tayrtahn) [reply Content.Server/Speech/Components/ListenWireAction.cs:44]: I assumed they didn't based on the other wire actions, but I have to admit that I didn't actually check!
I'll try it out and see

Comment (Tayrtahn) [reply Content.Server/Speech/Components/ListenWireAction.cs:44]: I assumed they didn't based on the other wire actions, but I have to admit that I didn't actually check!
I'll try it out and see.

Comment (Tayrtahn) [reply Content.Server/Speech/Components/ListenWireAction.cs:44]: Hm, it appears that they do not. I get null reference exceptions when trying to use the [Dependency] attribute instead.

File: Content.Server/VoiceMask/VoiceMaskComponent.cs [hunk 2]
```diff
@@ -6,4 +8,9 @@ public sealed partial class VoiceMaskComponent : Component
     [ViewVariables(VVAccess.ReadWrite)] public bool Enabled = true;
 
     [ViewVariables(VVAccess.ReadWrite)] public string VoiceName = "Unknown";
+
+    /// <summary>
+    /// If not null, overrides the speech verb used when this entity speaks.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)] public SpeechVerbPrototype? SpeechVerb;
```
Comment (EmoGarbage404) [inline Content.Server/VoiceMask/VoiceMaskComponent.cs:15]: you should be storing a ProtoId<SpeechVerbPrototype> and then indexing it when necessary. generally, you shouldn't be storing prototypes.

File: Content.Server/Speech/Components/ListenWireAction.cs [hunk 3]
```diff
@@ -7,18 +7,25 @@
...
 
     /// <summary>
     /// Length of the gibberish string sent when pulsing the wire
     /// </summary>
     private int _noiseLength = 16;
+
+    /// <summary>
+    /// Identifier of the SpeechVerbPrototype to use when pulsing the wire
+    /// </summary>
+    private string _speechVerb = "Electricity";
```
Comment (EmoGarbage404) [inline Content.Server/Speech/Components/ListenWireAction.cs:28]: should use ValidatePrototypeId<T> for this (also make it a constant)

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 4]
```diff
@@ -873,10 +877,13 @@ public sealed class TransformSpeakerNameEvent : EntityEventArgs
     public EntityUid Sender;
     public string Name;
 
-    public TransformSpeakerNameEvent(EntityUid sender, string name)
+    public SpeechVerbPrototype? SpeechVerb;
```
Comment (EmoGarbage404) [inline Content.Server/Chat/Systems/ChatSystem.cs:880]: like i mentioned before, you ideally shouldn't be passing around prototypes. Just get the ID and then index it when necessary. This also saves event subscribers from having to get PrototypeManager to index it.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by Tayrtahn** [comment]
Fixed. The other test failure seems to be widespread and unrelated.

File: Content.Server/VoiceMask/VoiceMaskComponent.cs [hunk 5]
```diff
@@ -6,4 +8,14 @@ public sealed partial class VoiceMaskComponent : Component
     [ViewVariables(VVAccess.ReadWrite)] public bool Enabled = true;
 
     [ViewVariables(VVAccess.ReadWrite)] public string VoiceName = "Unknown";
+
+    /// <summary>
+    /// If EnableSpeechVerbModification is true, overrides the speech verb used when this entity speaks.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [ValidatePrototypeId<SpeechVerbPrototype>]
+    public string SpeechVerb = "Default";
```
Comment (metalgearsloth) [inline Content.Server/VoiceMask/VoiceMaskComponent.cs:17]: Use ProtoId

File: Content.Server/VoiceMask/VoiceMaskComponent.cs [hunk 6]
```diff
@@ -6,4 +8,14 @@ public sealed partial class VoiceMaskComponent : Component
...
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    public bool EnableSpeechVerbModification = false;
 }
```
Comment (metalgearsloth) [inline Content.Server/VoiceMask/VoiceMaskComponent.cs:21]: Why aren't these datafields?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/VoiceMask/VoiceMaskComponent.cs (hunk 5 above)
Comment (Tayrtahn) [reply Content.Server/VoiceMask/VoiceMaskComponent.cs:17]: If I change it to a ProtoId, is it okay not to have the ValidatePrototypeId on it? I'd like to have it nullable and only use it if it's non-null, so I can get rid of EnableSpeechVerbModification.

File: Content.Server/VoiceMask/VoiceMaskComponent.cs (hunk 6 above)
Comment (Tayrtahn) [reply Content.Server/VoiceMask/VoiceMaskComponent.cs:21]: Huh, good point. They probably all should be, right?

**Review by EmoGarbage404 — APPROVED** [review]

**Comment by deltanedas** [comment]
> i will make pr for selecting any player accent (including silicons for trolling on binary) after this

:trollface:

**Pull request merged**